homeassistant/components/binary_sensor/mqtt.py

```diff
@@ -137,6 +144,21 @@ def state_message_received(topic, payload, qos):
                                 self._name, self._state_topic)
                 return
 
+            if (self._state and self._off_delay is not None):
+
+                def off_delay_listener(now):
+                    """Switch device off after a delay."""
+                    self._delay_listener = None
+                    self._state = False
+                    self.async_schedule_update_ha_state()
+
+                if self._delay_listener is not None:
+                    self._delay_listener()
+
+                self._delay_listener = evt.async_track_point_in_utc_time(
```

Comment:
`evt.async_call_later`

Reply:
Yes, but evt.async_call_later() is limited to a delay in seconds though, it's not too limiting?

Reply:
The internal timer only ticks every second anyway, so you wouldn't be able to get sub-second delays anyways

Reply:
Thanks for clarifying, fixed now.